core/server/src/main/java/alluxio/master/file/meta/InodeTree.java

```diff
@@ -102,7 +103,7 @@ public Object getFieldValue(Inode<?> o) {
   private final MountTable mMountTable;
 
   @SuppressWarnings("unchecked")
-  private final IndexedSet<Inode<?>> mInodes = new IndexedSet<>(ID_INDEX);
+  private FieldIndex<Inode<?>> mInodes = new UniqueFieldIndex<Inode<?>>(ID_INDEX);
```

Comment:
Also I suggest to add more comments here indicating why we use `UniqueFieldIndex` directly rather than `IndexedSet`